assets/model_monitoring/components/tests/e2e/utils/test_publish_components_for_ci.py

```diff
@@ -13,6 +13,6 @@
 class TestPublishComponentsForCI():
     """Class for model-monitoring-ci workflow to upload components and data once before splitting tests to runners."""
 
-    def test_publish_components(self, upload_component_version_file):
+    def test_publish_components(self, upload_component_version_file, cleanup_previous_e2e_tests):
```

Comment:
Does that mean every time one e2e test is running, the previous unfinished jobs will be cancelled? What if a new test is started before the previous test finish, would they have different experiment names?

Reply:
First see the comment above about the experiment names. Then, no this class module specifically is only ran once during CI setup and it publishes the components and now cleans up the tests. The issue rn is that every time you push a new commit to a PR or run the tests again if they failed it will trigger more e2e tests but the previous ones are still running. This can quickly overload the workspace and the tests will get stuck with not enough quota to run everything. We need to cleanup the previous tests

Reply:
I understand. My question is more like, let's see you have a PR and your PR is running a CI. Then I submitted another PR, which will also trigger a CI, and the cleanup will run. Will this cancel the jobs in your PR? I assume no, right? Only same PR will have same test_suite?